### PR TITLE
update README with native code architecture support in 5.1/2

### PR DESCRIPTION
**Pull request by avsm**
Following merge of #12276

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -12,9 +12,9 @@ the ports not yet supported in the 5.0 release you must use the 4.14 release.
 
 The initial release of OCaml 5.0 only supports the native compiler under ARM64
 and x86-64 architectures under Linux, macOS and the BSDs. On Windows, only the
-MinGW-w64 port is supported.  Support for other 64-bit architectures and
-systems will be added back in later releases. On 32-bit systems, only the
-bytecode compiler will be supported.
+MinGW-w64 port is supported. Native code support for Linux on RISC-V and
+s390x/IBM Z will be available in OCaml 5.1, and Linux on Power in OCaml 5.2.
+On 32-bit systems, only the bytecode compiler will be supported.
```
Comment (Octachron) [inline README.adoc:19]: We might want to mention that `Cygwin` support has been restored in 5.1 too.

Comment (avsm) [reply README.adoc:19]: Fixed in 21cf76b140744568713996445337da21105f1acb. Also moved the Cygwin changes entry over to the 'restored backends' part of the 5.1 Changes section.

File: README.adoc [hunk 2]
```diff
@@ -12,8 +12,10 @@ the ports not yet supported in the 5.0 release you must use the 4.14 release.
 
 The initial release of OCaml 5.0 only supports the native compiler under ARM64
 and x86-64 architectures under Linux, macOS and the BSDs. On Windows, only the
-MinGW-w64 port is supported. Native code support for Linux on RISC-V and
-s390x/IBM Z will be available in OCaml 5.1, and Linux on Power in OCaml 5.2.
+MinGW-w64 port is supported in OCaml 5.0 and Cygwin is restored in 5.1. On Linux,
```
Comment (Octachron) [inline README.adoc]: `the Cygwin port`? Otherwise, it sounds to me like something happened to `Cygwin` itself?

Comment (avsm) [reply README.adoc]: Fixed and rebased.

**Review by Octachron — APPROVED** [review]
Thanks for the documentation and Changes update !

**Pull request merged**